go/cmd/vtbackup/vtbackup.go

```diff
@@ -99,52 +100,62 @@ const (
-	initDBSQLFile = flag.String("init_db_sql_file", "", "path to .sql file to run after mysql_install_db")
-	detachedMode  = flag.Bool("detach", false, "detached mode - run backups detached from the terminal")
+	mysqlPort     = 3306
+	mysqlSocket   string
+	mysqlTimeout  = 5 * time.Minute
+	initDBSQLFile string
+	detachedMode  bool
 )
 
+func registerVtbackupFlags(fs *pflag.FlagSet) {
+	fs.DurationVar(&minBackupInterval, "min_backup_interval", minBackupInterval, "Only take a new backup if it's been at least this long since the most recent backup.")
+	fs.DurationVar(&minRetentionTime, "min_retention_time", minRetentionTime, "Keep each old backup for at least this long before removing it. Set to 0 to disable pruning of old backups.")
+	fs.IntVar(&minRetentionCount, "min_retention_count", minRetentionCount, "Always keep at least this many of the most recent backups in this backup storage location, even if some are older than the min_retention_time. This must be at least 1 since a backup must always exist to allow new backups to be made")
+	fs.BoolVar(&initialBackup, "initial_backup", initialBackup, "Instead of restoring from backup, initialize an empty database with the provided init_db_sql_file and upload a backup of that for the shard, if the shard has no backups yet. This can be used to seed a brand new shard with an initial, empty backup. If any backups already exist for the shard, this will be considered a successful no-op. This can only be done before the shard exists in topology (i.e. before any tablets are deployed).")
+	fs.BoolVar(&allowFirstBackup, "allow_first_backup", allowFirstBackup, "Allow this job to take the first backup of an existing shard.")
+	fs.BoolVar(&restartBeforeBackup, "restart_before_backup", restartBeforeBackup, "Perform a mysqld clean/full restart after applying binlogs, but before taking the backup. Only makes sense to work around xtrabackup bugs.")
+	// vttablet-like flags
+	fs.StringVar(&initDbNameOverride, "init_db_name_override", initDbNameOverride, "(init parameter) override the name of the db used by vttablet")
+	fs.StringVar(&initKeyspace, "init_keyspace", initKeyspace, "(init parameter) keyspace to use for this tablet")
+	fs.StringVar(&initShard, "init_shard", initShard, "(init parameter) shard to use for this tablet")
+	fs.IntVar(&concurrency, "concurrency", concurrency, "(init restore parameter) how many concurrent files to restore at once")
+	// mysqlctld-like flags
+	fs.IntVar(&mysqlPort, "mysql_port", mysqlPort, "mysql port")
+	fs.StringVar(&mysqlSocket, "mysql_socket", mysqlSocket, "path to the mysql socket")
+	fs.DurationVar(&mysqlTimeout, "mysql_timeout", mysqlTimeout, "how long to wait for mysqld startup")
+	fs.StringVar(&initDBSQLFile, "init_db_sql_file", initDBSQLFile, "path to .sql file to run after mysql_install_db")
+	fs.BoolVar(&detachedMode, "detach", detachedMode, "detached mode - run backups detached from the terminal")
+}
+
+func init() {
+	servenv.OnParseFor("vtbackup", registerVtbackupFlags)
```

Comment:
`OnParseFor` => `OnParse`

Reply:
The reason i use dOnParseFor, is OnParse is adding in globalhook and I thought that might not be what we intend here .. we just need that flag to vtbackup but not for all the command right?

Reply:
nothing can import another `package main`, so _anything_ registered here is by definition "global" (because this package is the entire "globe", if you will)